backend/src/main/java/in/kahl/promptwhispers/model/Game.java

```diff
@@ -1,30 +1,28 @@
 package in.kahl.promptwhispers.model;
 
-import in.kahl.promptwhispers.Renderable;
 import org.springframework.data.annotation.Id;
 
 import java.time.Instant;
 import java.util.Collections;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.LinkedList;
+import java.util.List;
 import java.util.UUID;
 
 public record Game(
         @Id
         String id,
-        Map<Integer, Renderable> steps,
+        List<Step> steps,
         Instant createdAt,
         Boolean isFinished
 ) {
     public Game() {
-        this(UUID.randomUUID().toString(), Collections.emptyMap(), Instant.now(), false);
+        this(UUID.randomUUID().toString(), Collections.emptyList(), Instant.now(), false);
     }
 
-    public Game withStep(Renderable step) {
-        HashMap<Integer, Renderable> map = new HashMap<>(steps());
+    public Game withStep(Step step) {
+        List<Step> stepList = new LinkedList<>(steps());
+        stepList.addLast(step);
```

Comment:
Wieso benutzt du hier eine LinkedList? Ist das notwendig? Eine ArrayList behält ja trotzdem die order of insertion :)